joern-cli/frontends/kotlin2cpg/src/main/scala/io/joern/kotlin2cpg/ast/KtPsiToAst.scala

```diff
@@ -790,61 +787,182 @@ trait KtPsiToAst {
     }
   }
 
-  // TODO: clean up this whole fn
-  def astForQualifiedExpression(expr: KtQualifiedExpression, argIdx: Option[Int])(implicit
+  private def astForQualifiedExpressionFieldAccess(expr: KtQualifiedExpression, argIdx: Option[Int])(implicit
     typeInfoProvider: TypeInfoProvider
   ): Ast = {
-    val callKind        = typeInfoProvider.bindingKind(expr)
-    val isStaticCall    = callKind == CallKinds.StaticCall
-    val isDynamicCall   = callKind == CallKinds.DynamicCall
-    val isExtensionCall = callKind == CallKinds.ExtensionCall
+    val receiverAst = astsForExpression(expr.getReceiverExpression, Some(1)).head
```

Comment:
This is a possibly unsafe use of `head`. Given valid code, this should never crash, but something like `headOption.getOrElse(someDefault)` like you used above is safer.